### ISSUE TITLE
NoMethodError - undefined method `user_targets' for #<Pod::Installer::PostInstallHooksContext::UmbrellaTargetDescription:0x007fcc968c23b8>

### ISSUE THREAD

**Issue by lyp944** [CocoaPods/CocoaPods#4741]
# NoMethodError - undefined method `user_targets' 
##my Podfile
```platform :ios, '8.0'
inhibit_all_warnings!

source 'https://github.com/CocoaPods/Specs.git'

pod 'AFNetworking', '~> 3.0'
pod 'SDWebImage'
pod 'UIActivityIndicator-for-SDWebImage'
pod 'Mantle'
pod 'Aspects'
#pod 'ReactiveCocoa', '2.1.8'
pod 'WebViewJavascriptBridge'
pod 'pop'
pod 'SSKeychain'
pod 'SVPullToRefresh'
pod 'FDFullscreenPopGesture'
pod 'YYKit'
pod 'Masonry'
pod 'MJRefresh'
```
##ruby log
### Error

```
NoMethodError - undefined method `user_targets' for #<Pod::Installer::PostInstallHooksContext::UmbrellaTargetDescription:0x007fcc968c23b8>
/Users/liyp/.rvm/gems/ruby-2.0.0-p643/gems/cocoapods-stats-1.0.0.beta.2/lib/cocoapods_stats/target_mapper.rb:20:in `block in pods_from_project'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643/gems/cocoapods-stats-1.0.0.beta.2/lib/cocoapods_stats/target_mapper.rb:9:in `each'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643/gems/cocoapods-stats-1.0.0.beta.2/lib/cocoapods_stats/target_mapper.rb:9:in `flat_map'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643/gems/cocoapods-stats-1.0.0.beta.2/lib/cocoapods_stats/target_mapper.rb:9:in `pods_from_project'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-stats-0.6.2/lib/cocoapods_plugin.rb:31:in `block (2 levels) in <module:CocoaPodsStats>'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/lib/cocoapods/user_interface.rb:80:in `titled_section'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-stats-0.6.2/lib/cocoapods_plugin.rb:27:in `block in <module:CocoaPodsStats>'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/lib/cocoapods/hooks_manager.rb:117:in `call'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/lib/cocoapods/hooks_manager.rb:117:in `block (3 levels) in run'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/lib/cocoapods/user_interface.rb:140:in `message'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/lib/cocoapods/hooks_manager.rb:111:in `block (2 levels) in run'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/lib/cocoapods/hooks_manager.rb:109:in `each'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/lib/cocoapods/hooks_manager.rb:109:in `block in run'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/lib/cocoapods/user_interface.rb:140:in `message'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/lib/cocoapods/hooks_manager.rb:108:in `run'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/lib/cocoapods/installer.rb:481:in `run_plugins_post_install_hooks'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/lib/cocoapods/installer.rb:462:in `perform_post_install_actions'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/lib/cocoapods/installer.rb:113:in `install!'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/lib/cocoapods/command/project.rb:71:in `run_install_with_update'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/lib/cocoapods/command/project.rb:101:in `run'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/claide-0.9.1/lib/claide/command.rb:312:in `run'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/lib/cocoapods/command.rb:47:in `run'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643@global/gems/cocoapods-0.39.0/bin/pod:44:in `<top (required)>'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643/bin/pod:23:in `load'
/Users/liyp/.rvm/gems/ruby-2.0.0-p643/bin/pod:23:in `<main>'
```
##HELP
WHY? What can I do to  fix it?

**Comment by segiddins** [comment]
CocoaPods 0.39 shouldn't be using the 1.0 version of CocoaPods stats...

**Comment by lyp944** [comment]
I have reduce the version number ，but it does not work.
```
sudo gem install cocoapods --version 0.38.0
```

 What settings should I use? Please show me the setting.Thank you very much.

**Comment by segiddins** [comment]
What's the output of `gem env` for you?

**Comment by lyp944** [comment]
```*** LOCAL GEMS ***

activesupport (4.2.4)
bigdecimal (1.2.0)
claide (1.0.0.beta.1, 0.9.1)
cocoapods (1.0.0.beta.2, 0.39.0, 0.38.0)
cocoapods-core (1.0.0.beta.2, 0.39.0, 0.38.2, 0.38.0)
cocoapods-deintegrate (1.0.0.beta.1)
cocoapods-downloader (1.0.0.beta.1, 0.9.3)
cocoapods-plugins (1.0.0.beta.1, 0.4.2)
cocoapods-search (1.0.0.beta.1, 0.1.0)
cocoapods-stats (1.0.0.beta.2, 0.6.2, 0.5.3)
cocoapods-trunk (1.0.0.beta.1, 0.6.4)
cocoapods-try (1.0.0.beta.2, 0.5.1, 0.4.5)
colored (1.2)
escape (0.0.4)
fourflusher (0.3.0)
fuzzy_match (2.0.4)
gem-wrappers (1.2.7)
i18n (0.7.0)
io-console (0.4.2)
json (1.7.7)
minitest (5.8.0, 4.3.2)
molinillo (0.4.1, 0.4.0, 0.3.1)
nap (1.0.0, 0.8.0)
netrc (0.7.8)
psych (2.0.0)
rake (0.9.6)
rdoc (4.0.0)
rvm (1.11.3.9)
test-unit (2.0.0.0)
thread_safe (0.3.5)
tzinfo (1.2.2)
xcodeproj (1.0.0.beta.2, 0.28.2, 0.26.3)
```

**Comment by segiddins** [comment]
@alloy I think this is caused by the old, naive CLAide plugin manager?

**Comment by muhasturk** [comment]
I have similur issue.  Full output

```
 pod install
Updating local specs repositories

CocoaPods 1.0.0.beta.2 is available.
To update use: `sudo gem install cocoapods --pre`
[!] This is a test version we'd love you to try.

For more information see http://blog.cocoapods.org
and the CHANGELOG for this version http://git.io/BaH8pQ.

Analyzing dependencies

――― MARKDOWN TEMPLATE ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

### Command

```
/usr/local/bin/pod install
```

### Report

* What did you do?

* What did you expect to happen?

* What happened instead?


### Stack

```
   CocoaPods : 0.39.0
        Ruby : ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-darwin15]
    RubyGems : 2.5.1
        Host : Mac OS X 10.11.2 (15C50)
       Xcode : 7.2 (7C68)
         Git : git version 2.7.0
Ruby lib dir : /usr/local/Cellar/ruby/2.3.0/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 2bf04bbb5041cc4e341f02a9fccc3a5255ef478c
```

### Plugins

```
cocoapods-plugins : 0.4.2
cocoapods-search  : 0.1.0
cocoapods-stats   : 0.6.2
cocoapods-trunk   : 0.6.4
cocoapods-try     : 0.5.1
```

### Podfile

```ruby
platform :ios, '8.0'
use_frameworks!

target 'Kidofun' do
    pod 'Fabric'
    pod 'Crashlytics'
end

target 'KidofunTests' do

end
```

### Error

```
NoMethodError - undefined method `to_ary' for #<Pod::Specification name="Crashlytics">
Did you mean?  to_query
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/resolver/lazy_specification.rb:14:in `method_missing'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/resolver.rb:64:in `flatten'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/resolver.rb:64:in `block in resolve'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/resolver.rb:63:in `tap'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/resolver.rb:63:in `resolve'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/installer/analyzer.rb:539:in `block in resolve_dependencies'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/user_interface.rb:59:in `section'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/installer/analyzer.rb:537:in `resolve_dependencies'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/installer/analyzer.rb:70:in `analyze'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/installer.rb:213:in `analyze'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/installer.rb:136:in `block in resolve_dependencies'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/user_interface.rb:59:in `section'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/installer.rb:135:in `resolve_dependencies'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/installer.rb:105:in `install!'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/command/project.rb:71:in `run_install_with_update'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/command/project.rb:101:in `run'
/usr/local/lib/ruby/gems/2.3.0/gems/claide-0.9.1/lib/claide/command.rb:312:in `run'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/command.rb:47:in `run'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/bin/pod:44:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```

――― TEMPLATE END ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

[!] Oh no, an error occurred.

Search for existing GitHub issues similar to yours:
https://github.com/CocoaPods/CocoaPods/search?q=undefined+method+%60to_ary%27+for+%23%3CPod%3A%3ASpecification+name%3D%22Crashlytics%22%3E%0ADid+you+mean%3F++to_query&type=Issues

If none exists, create a ticket, with the template displayed above, on:
https://github.com/CocoaPods/CocoaPods/issues/new

Be sure to first read the contributing guide for details on how to properly submit a ticket:
https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md

Don't forget to anonymize any private data!
```

**Comment by msfeldstein** [comment]
I was able to resolve this by running `gem uninstall cocoapods-stats`.  I had a beta version of the stats but not using beta version of cocoapods

**Comment by muhasturk** [comment]
In my opinion this issue relevant https://github.com/CocoaPods/CocoaPods/pull/4368/files because of new ruby.

**Comment by segiddins** [comment]
Nah, this is due to the way the CLAide plugin manager used to work. I have a fix in stats locally that I'll try and push up later.

**Comment by slizeray** [comment]
I just have created issue #4747 I guess this is the same issue? I can't use anymore cocoa pods, how could I recreate a working environment? Thanks

**Comment by segiddins** [comment]
For the moment, either install the beta and run with `pod _1.0.0.beta.2_` or uninstall the beta version of CocoaPods stats or wait until we release a fix.

**Comment by slizeray** [comment]
Thanks it works great with the following command:
pod _1.0.0.beta.2_ install

**Comment by segiddins** [comment]
This will be fixed by shipping https://github.com/CocoaPods/cocoapods-stats/issues/27

**Comment by narirou** [comment]
Please fix this issue... :pray:

**Comment by efremidze** [comment]
:+1:

**Comment by reduxdj** [comment]
Same here, I can't build my app, which means i can't submit it and I am blocked, suggestions, clear guidelines to fix ASAP would be greatly appreciated

**Comment by muhasturk** [comment]
you can use beta version to fix this issue temporarily

**Comment by segiddins** [comment]
You can run `gem install cocoapods-stats --pre` to fix this now :)

**Comment by Eliothu** [comment]
I have the same issue: when I run
 ```
EliotdeMacBook-Pro:~ eliot$ gem install cocoapods-stats --pre
Successfully installed cocoapods-stats-1.0.0.beta.3
ERROR:  While executing gem ... (Errno::EACCES)
Permission denied @ dir_s_mkdir - /usr/local/lib/ruby/gems/2.3.0/doc/cocoapods-stats-1.0.0.beta.3
```
The I run ` pod update `
### Error

```
NoMethodError - undefined method `to_ary' for #<Pod::Specification name="AFNetworking">
Did you mean?  to_query
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/resolver/lazy_specification.rb:14:in `method_missing'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/resolver.rb:64:in `flatten'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/resolver.rb:64:in `block in resolve'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/resolver.rb:63:in `tap'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/resolver.rb:63:in `resolve'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/installer/analyzer.rb:539:in `block in resolve_dependencies'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/user_interface.rb:59:in `section'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/installer/analyzer.rb:537:in `resolve_dependencies'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/installer/analyzer.rb:70:in `analyze'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/installer.rb:213:in `analyze'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/installer.rb:136:in `block in resolve_dependencies'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/user_interface.rb:59:in `section'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/installer.rb:135:in `resolve_dependencies'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/installer.rb:105:in `install!'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/command/project.rb:71:in `run_install_with_update'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/command/project.rb:156:in `run'
/usr/local/lib/ruby/gems/2.3.0/gems/claide-0.9.1/lib/claide/command.rb:312:in `run'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/lib/cocoapods/command.rb:47:in `run'
/usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-0.39.0/bin/pod:44:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```
There is something do to can I fix this problem?  thanks a lot.

**Comment by segiddins** [comment]
If you're using ruby 2.3, you'll have to use the CocoaPods beta.

**Comment by Eliothu** [comment]
Thanks , I fix it when I update 1.0.0.beta.

**Comment by Eliothu** [comment]
<img width="707" alt="2016-01-27 12 20 13" src="https://cloud.githubusercontent.com/assets/12962550/12603764/7964b232-c4f0-11e5-93a8-31e9a3c643e0.png">
<img width="250" alt="2016-01-27 12 20 31" src="https://cloud.githubusercontent.com/assets/12962550/12603779/8cf51594-c4f0-11e5-85dc-0695b40c599e.png">

When I build myproject.xcworkspace there is some problem looks like above. 

```
source 'https://github.com/CocoaPods/Specs.git'
platform :ios, '7.0'
inhibit_all_warnings!

target 'jiuhuar' do

pod 'AFNetworking', '~> 3.0.0-beta.3'

pod 'ReactiveCocoa'

pod 'Masonry'

pod 'SDWebImage'

pod 'MJRefresh'

pod 'MJExtension'

pod 'MBProgressHUD'

pod 'FMDB'

pod 'TTTAttributedLabel'

pod "Qiniu", :git => 'https://github.com/qiniu/objc-sdk.git', :branch => 'AFNetworking-3.x'

pod 'YYKit'

end
```
The target is correct ? 
Please help or try to give some ideas how to achieve this.
